03-rbac/proposal.md

```diff
@@ -0,0 +1,159 @@
+# Summary
+
+Introduces a proposal for RBAC. Specifically:
+
+- What roles do we need?
+- How would fly set-team work?
+- How do roles get persisted in the database?
+- How are roles handled by the API?
+
+
+# Proposal
+
+## What roles do we need?
```

Comment:
Here are our needs, at a minimum:
- dashboard - we have a custom dashboard. It needs GET access to the following and no more, so that the credentials that the dashboard uses can't be used to get build logs that could leak other credentials.
  - `/api/v1/teams/{team}/pipelines`
  - `/api/v1/teams/{team}/pipelines/{pipeline}/jobs`
  - `/api/v1/teams/{team}/pipelines/{pipeline}/resources` (to display check failures)
  - `/api/v1/teams/{team}/pipelines`
  - `/api/v1/teams/{team}/builds`
- user - everything in their team, except `set-team`, `hijack`, `execute`, `set-pipeline`. We want to ensure every action that concourse does is auditable. We have a pipeline that gets pipeline configs from git, and does the set-pipeline-ing, and we need to ensure that is the only way that pipelines can be set
- admin - everything

Because these roles and permissions are quite a bit different, and I'm sure others will have their own idea of what they should be, it would be great if this was configurable.